mlblocks/mlpipeline.py

```diff
@@ -154,6 +154,45 @@ def _get_block_outputs(self, block_name):
 
         return outputs
 
+    def _get_block_outputs_dict(self, block_name):
```

Comment:
I think that we might be repeating too much code. Once these methods are added we will end up having 4 methods which are almost the same. Yet, they still have slight differences, so it's a tricky situation.

I suggest the following:

1. Create a generic method `_get_block_variables`, which receives 3 string arguments: `block_name`, `variables_attr`, `names`.
    - `block_name (str)`: name of the block for which to get the io specification
    - `variables_attr (str)`: Name of the attribute that has the variables list. It can be `fit_args`, `produce_args` or `produce_output`.
    - `names (dict)`: Dictionary used to translate the variable names. We will pass `self.output_names` or `self.input_names`.

This method will basically do the same as the three that you added, but using `variables = getattr(block, variables_attr)` to get the list of inputs/outputs, and use `names` instead of `self.output_names` or `self.input_names`.

2. Modify `_get_block_outputs` so that it uses the method above instead of repeating the loop:

```python
def _get_block_outputs(self, block_name):
    outputs = _get_block_variables(block_name, 'produce_output', self.output_names)
    for context_name, output in outputs.items():
        output['variable'] = '{}.{}'.format(block_name, context_name)

    return list(outputs.values())
```

3. Modify the `_get_inputs` code so it uses the new method instead of the old ones.